dmidecode/struct_parser.go

```diff
@@ -164,27 +165,50 @@ func parseStruct(t *smbios.Table, off int, complete bool, sp interface{}) (int,
 	return off, nil
 }
 
-func toTable(t *smbios.Table, sp interface{}) (int, error) {
+// Table is an SMBIOS table.
+type Table interface {
+	Typ() smbios.TableType
+}
+
+const headerLen = 4
+
+// ToTable converts a struct to an smbios.Table.
+func ToTable(val Table, handle uint16) (*smbios.Table, error) {
+	t := smbios.Table{
+		Header: smbios.Header{
+			Type:   val.Typ(),
+			Handle: handle,
```

Comment:
why we have to pass in `handle` when we can directly obtain that from  `val.Handle`?